src/main/java/com/thebois/views/game/TerrainView.java

```diff
@@ -1,23 +1,38 @@
 package com.thebois.views.game;
 
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
 
 import com.badlogic.gdx.graphics.Color;
 import com.badlogic.gdx.graphics.Texture;
 import com.badlogic.gdx.graphics.g2d.Batch;
 
-import com.thebois.models.world.ITerrain;
+import com.thebois.models.world.terrains.ITerrain;
+import com.thebois.models.world.terrains.TerrainType;
 import com.thebois.views.TextureUtils;
 
 /**
  * World view handles all the drawing of the world itself and its information on how to draw it.
  */
 public class TerrainView implements IView {
 
-    private final Color grassColor = new Color(0.005f, 0.196f, 0.107f, 1);
+    private static final Color GRASS_COLOR = Color.valueOf("#008000");
+    private static final Color DIRT_COLOR = Color.valueOf("#76552b");
+    private static final Color SAND_COLOR = Color.valueOf("#bbc201");
+    private static final Map<TerrainType, Color> TERRAIN_COLOR;
+
+    static {
+        TERRAIN_COLOR = new HashMap<>();
+        TERRAIN_COLOR.put(TerrainType.GRASS, GRASS_COLOR);
+        TERRAIN_COLOR.put(TerrainType.DIRT, DIRT_COLOR);
+        TERRAIN_COLOR.put(TerrainType.SAND, SAND_COLOR);
```

Comment:
As I go trough this review I see a reoccurring theme where we have to update this everywhere. And as you guess I am wondering if we can minimize the places we have to update?

Reply:
Well, since this is a view you'll always have to change this if you add something in the model that you want to render differently.

Unless you're suggesting we put colors in the model? 😛

Reply:
So we basically have an enum that mirrors the model but with colors and textures instead?

Reply:
Made the enums, take a look and tell me what you think.

Imo Resource and Terrain view is a lot cleaner, but maybe not the easiest to read enum but at least everything is in the same place.